### PR TITLE
Add access zone image

### LINKED ISSUE #38: Build access zone image
Objective: to have a zone image that is configurable using mdata.

Desirable features:
- routing enabled
- DHCP client enabled on all interfaces
- setup firewall + nat
- mapping ports to inside hosts (reverse nat)

Bonus features:
- OpenVPN
- integrated EasyRSA
- capability of CARP/VRRP
- Zabbix proxy

Can be based on this howto https://docs.danubecloud.org/user-guide/howto/access_zone.html

It can be hardcoded that net0 is the WAN interface.

Proposal of mdata variables ('x' is the number of the network interface):
- NAT_ENABLE - default: true. If false, IP addresses will be forwarded without NAT. It is on maintainer/creators's decision if following reverse nat settings will be applicable if NAT_ENABLE=false.
- REVERSE_NAT_RULES_TCP - either json or comma separated pairs. These ports on WAN interface will be forwarded to internal hosts. Format: external_port:internal_ip_address[:dst_port],external_port2:internal_ip_address2[:dst_port2] e.g. "80:10.0.0.100:80,443:10.0.0.100:443,53:10.0.0.102:53
- REVERSE_NAT_RULES_UDP
- NETx_SOURCES_ALLOW - default: any. Format: comma separated list of hosts or subnets, e.g. 44.33.22.11,50.60.70.0/24
- NETx_ALLOWED_PORTS_TCP - default: any. Format: comma separated list of port numbers
- NETx_ALLOWED_PORTS_UDP
- OPENVPN_ENABLE - default: false
- OPENVPN_CA_NAME - generate self-signed certificate for creating openvpn client certificates. New user certificates can be created using easyrsa script (instructions how to do it can be put into /etc/motd)
- NETx_CARP_ENABLE - default: false
- NETx_CARP_VIP
- NETx_CARP_WEIGHT

### PR DESCRIPTION
**Pull request by YanChii**
Not very featureful so far.. but serves the main purpose - create a NAT zone for any subnet in one click - closes #38

### CODE REVIEW

File: ansible/build-access-zone.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    - zabbix-agent
+    - access-zone
+    - vm-image
+
+- name: Create and save image
+  hosts: buildnode
+  vars_files:
+    - vars/build/vm/access-zone.yml
+  vars:
+    image_tags: {internal: true, resize: true, deploy: false}
```
Comment (dn0) [inline ansible/build-access-zone.yml]: internal=false

File: ansible/vars/build/vm/access-zone.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+image_desc: "Dedicated router zone"
+image_homepage: "{{ image_base_homepage }}#access-zone"
+remove_base_image: true
+builder_dir: "{{ builder.appliance.dir }}/access-zone"
+zone_brand: joyent
+zone_dns_domain: local
+zone_uuid: 4cb74eb8-8f37-4a0d-aa37-7d8f50083970
+zone_image_uuid: "{{ (lookup('pipe', 'curl -s {{ builder_base_zone_imgmanifest_url }}') | from_json)['uuid'] }}"
+zone_hostname: access-zone
+zone_nic_0_ip: "{{ build_ips.esdc_dns | default(build_ip) }}"
```
Comment (dn0) [inline ansible/vars/build/vm/access-zone.yml]: ~~esdc_dns~~

File: docs/network/access-zone.rst [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+* Filter: ``/etc/ipf/ipf.conf``
+* NAT and port mapper: ``/etc/ipf/ipnat.conf``
+
+In order to use it, you should create a VM in a following way:
+* The net0 network interface (the first one) has to be a WAN interface (all outgoing traffic from this interface will be masqueraded)
+* All other network interfaces need to have the *Allow IP Spoofing* option enabled (otherwise all routed packets will be dropped)
+
+.. note:: A virtual machine using this image requires at least 256 MB of RAM.
+
+The image supports following metadata:
```
Comment (dn0) [inline docs/network/access-zone.rst]: Please add information (link to) that it is based on base-64-es.

File: docs/network/access-zone.rst [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+* **root_authorized_keys**: content for ``/root/.ssh/authorized_keys``.
+* **cloud-init** modules: growpart, resizefs, ssh, set-passwords.
+* **org.erigones:zabbix_ip**: ``Server`` and ``ServerActive`` parameters zabbix_agentd.conf.
+
+
+Changelog
+---------
+
+2.6.1
```
Comment (dn0) [inline docs/network/access-zone.rst]: Are we going to use this version scheme, or dates?

File: docs/network/access-zone.rst [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+In order to use it, you should create a VM in a following way:
+* The net0 network interface (the first one) has to be a WAN interface (all outgoing traffic from this interface will be masqueraded)
+* All other network interfaces need to have the *Allow IP Spoofing* option enabled (otherwise all routed packets will be dropped)
+
+.. note:: A virtual machine using this image requires at least 256 MB of RAM.
+
+The image supports following metadata:
+
+* **root_authorized_keys**: content for ``/root/.ssh/authorized_keys``.
+* **cloud-init** modules: growpart, resizefs, ssh, set-passwords.
```
Comment (dn0) [inline docs/network/access-zone.rst]: No cloud-init here.

File: docs/network/access-zone.rst [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+`Dedicated router image for `Danube Cloud <https://danubecloud.org>`__.
+
+This is a SunOS zone image with enabled packet forwarding, firewall and NAT. It supports initialization through `Zoneinit <https://github.com/joyent/zoneinit>`__.
+
+Firewall configuration can be altered here:
+* Filter: ``/etc/ipf/ipf.conf``
+* NAT and port mapper: ``/etc/ipf/ipnat.conf``
+
+In order to use it, you should create a VM in a following way:
```
Comment (dn0) [inline docs/network/access-zone.rst]: VM -> SunOS Zone (thats the OS type)

**Review by dn0 — CHANGES_REQUESTED** [review]

**Comment by YanChii** [comment]
Is it ok to merge now?

File: ansible/vars/build/vm/network-access-zone.yml [hunk 7]
```diff
@@ -0,0 +1,22 @@
+#Dependencies: base.yml, zone.yml
+image_name: network-access-zone
+image_desc: "Dedicated router/NAT/VPN zone container"
+image_homepage: "{{ image_network_homepage }}/access-zone.rst"
+remove_base_image: true
+builder_dir: "{{ builder.appliance.dir }}/network-access-zone"
+zone_brand: joyent
+zone_dns_domain: local
+zone_uuid: 4cb74eb8-8f37-4a0d-aa37-7d8f50083970
```
Comment (dn0) [inline ansible/vars/build/vm/network-access-zone.yml]: Please change this uuid to something unique.

**Review by dn0 — CHANGES_REQUESTED** [review]

File: ansible/vars/build/vm/network-access-zone.yml (hunk 7 above)
Comment (YanChii) [reply ansible/vars/build/vm/network-access-zone.yml]: Done.

**Review by dn0 — APPROVED** [review]

**Pull request merged**